plugin/Model/FormModelBuilder/FormModelBuilderSimilarEstateSettings.php

```diff
@@ -0,0 +1,374 @@
+		$pInputModelFieldsConfig->setValuesAvailable($fieldNames);
+		$pInputModelFieldsConfig->setValue($fields);
+		return $pInputModelFieldsConfig;
+	}
+
+	/**
+	 * @param string $field
+	 * @return InputModelOption
+	 * @throws ExceptionInputModelMissingField
+	 */
+	public function createInputModelTemplate(string $field = InputModelOptionFactorySimilarView::INPUT_TEMPLATE)
+	{
+		$labelTemplate = __('Template', 'onoffice');
+		$pInputModelTemplate = $this->_pInputModelSimilarViewFactory->create($field, $labelTemplate);
+		$pInputModelTemplate->setHtmlType(InputModelOption::HTML_TYPE_SELECT);
+		$pInputModelTemplate->setValuesAvailable($this->readTemplatePaths('estate'));
+		$pInputModelTemplate->setValue($this->getTemplateValueByField($field));
+
+		return $pInputModelTemplate;
+	}
+
+
+	/**
+	 *
+	 * @param string $field
+	 * @return string
+	 *
+	 */
+
+	private function getTemplateValueByField(string $field): string
+	{
+		switch ($field) {
+			case InputModelOptionFactorySimilarView::INPUT_FIELD_SIMILAR_ESTATES_TEMPLATE:
+				return $this->_pDataSimilarView->getDataViewSimilarEstates()->getTemplate();
+			default:
+				return '';
+		}
+	}
+
+
+	/**
+	 *
+	 * @return InputModelDB
+	 *
+	 */
+
+	public function createInputModelSimilarEstateKind()
+	{
+		$pDataViewSimilarEstates = $this->_pDataSimilarView->getDataViewSimilarEstates();
+
+		$labelSameKind = __('Same Kind of Estate', 'onoffice');
+
+		$pInputModelSimilarEstateKind = $this->_pInputModelSimilarViewFactory->create
+			(InputModelOptionFactorySimilarView::INPUT_FIELD_SIMILAR_ESTATES_SAME_KIND, $labelSameKind);
+		$pInputModelSimilarEstateKind->setHtmlType(InputModelOption::HTML_TYPE_CHECKBOX);
+
+		$pInputModelSimilarEstateKind->setValuesAvailable(1);
+		$pInputModelSimilarEstateKind->setValue($pDataViewSimilarEstates->getSameEstateKind());
+
+		return $pInputModelSimilarEstateKind;
+	}
+
+
+	/**
+	 *
+	 * @return InputModelDB
+	 *
+	 */
+
+	public function createInputModelSimilarEstateMarketingMethod()
+	{
+		$pDataViewSimilarEstates = $this->_pDataSimilarView->getDataViewSimilarEstates();
+
+		$labelSameMarketingMethod = __('Same Marketing Method', 'onoffice');
+
+		$pInputModelSameMarketingMethod = $this->_pInputModelSimilarViewFactory->create
+			(InputModelOptionFactorySimilarView::INPUT_FIELD_SIMILAR_ESTATES_SAME_MARKETING_METHOD, $labelSameMarketingMethod);
+		$pInputModelSameMarketingMethod->setHtmlType(InputModelOption::HTML_TYPE_CHECKBOX);
+
+		$pInputModelSameMarketingMethod->setValuesAvailable(1);
+		$pInputModelSameMarketingMethod->setValue($pDataViewSimilarEstates->getSameMarketingMethod());
+
+		return $pInputModelSameMarketingMethod;
+	}
+
+    /**
+     *
+     * @return InputModelDB
+     *
+     */
+
+    public function createInputModelDontShowArchived()
+    {
+        $pDataViewSimilarEstates = $this->_pDataSimilarView->getDataViewSimilarEstates();
+
+        $labelDontShowArchived= __('Don&#8217;t show archived estates', 'onoffice');
```

Comment:
Text within `__()` should be plain and not HTML entity encoded. Encoding and escaping should happing where output starts.